### PR TITLE
Text Processing Refactoring

### PR DESCRIPTION
**Pull request by LinYunMo**
Re:  https://github.com/cocos/3d-tasks/issues/15357

Restructure and add text processing

### CODE REVIEW

File: cocos/2d/assembler/label/text-process-data.ts [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    public isItalic = false; // ttf
+    // under line // style
+    public isUnderline = false; // ttf
+    public underlineHeight = 1; // ttf
+    // outline style
+    public isOutlined = false; // both // ttf & char
+    public outlineColor = Color.WHITE.clone(); // both // ttf & char
+    public outlineWidth = 1; // both // ttf & char
+    // shadowStyle
+    public hasShadow = false; // ttf
```
Comment (SantyWang) [inline cocos/2d/assembler/label/text-process-data.ts]: Is these parameters affecting text layout?

File: cocos/2d/assembler/label/text-process-data.ts [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    public canvasSize = new Size(); // ttf
+
+    public canvasPadding = new Rect(); // ttf
+    public contentSizeExtend = Size.ZERO.clone(); // ttf
+
+    public startPosition = Vec2.ZERO.clone(); // ttf
+
+    // -----------------------bitMap extra part-------------------------
+
+    public fntConfig: IConfig | null = null;
```
Comment (SantyWang) [inline cocos/2d/assembler/label/text-process-data.ts]: Maybe pass font directly ?

File: cocos/2d/assembler/label/text-process-data.ts [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    public fontDesc = ''; // both
+    public fontSize = 40; // input fonSize // both
+    public actualFontSize = 0; // both
+    // public declare _material: Material | null;
+
+    public color = Color.WHITE.clone(); // both
+
+    // Layout
+    // alignment
+    public hAlign = 0;// Enum  // both
```
Comment (SantyWang) [inline cocos/2d/assembler/label/text-process-data.ts]: Should split rendering and layout data?

File: cocos/2d/assembler/label/text-process-data.ts [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+    // node part
+    public nodeContentSize = Size.ZERO.clone(); // both
+    // anchor
+    public uiTransAnchorX = 0.5; // both
+    public uiTransAnchorY = 0.5; // both
+
+    // ---------------ttf extra part-----------------
+    // bold // style
+    public isBold = false; // ttf
```
Comment (SantyWang) [inline cocos/2d/assembler/label/text-process-data.ts]: Should split a style data.

File: cocos/2d/assembler/label/text-process-data.ts [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+    // Process Output
+    public quadCount = 0; // both
+    public vertexBuffer: IRenderData[] = []; // both
+    public texture: Texture2D | SpriteFrame | null = null;  // both
+
+    // string before process
+    public inputString = ''; // both
+    // string after process
+    public parsedString: string[] = [];
```
Comment (SantyWang) [inline cocos/2d/assembler/label/text-process-data.ts]: Should split output and input data.

File: cocos/2d/assembler/label/text-process-data.ts [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+    public fontAtlas: FontAtlas | LetterAtlas | null = null;
+    public spriteFrame: SpriteFrame | null = null;
+
+    public spacingX = 0;
+    public originFontSize = 0;
+
+    public bmfontScale = 1.0;
+
+    // bmfont used temp value
+    public labelWidth = 0;
```
Comment (SantyWang) [inline cocos/2d/assembler/label/text-process-data.ts]: Should split a TextLayout to store these data.

File: cocos/2d/assembler/label/text-process-data.ts [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+import { SpriteFrame } from '../../assets';
+import { FontAtlas, IConfig } from '../../assets/bitmap-font';
+import { LetterAtlas } from './font-utils';
+import { TextStyle } from './text-style';
+import { TextLayout } from './text-layout';
+import { TextOutputLayoutData, TextOutputRenderData } from './text-output-data';
+
+export class TextProcessData {
+    get style () {
+        return this._style;
```
Comment (SantyWang) [inline cocos/2d/assembler/label/text-process-data.ts]: 为什么不直接多传几个参数，不用合并到一个参数里面

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Review by SantyWang — APPROVED** [review]

**Pull request merged**